dnf/base.py

```diff
@@ -319,6 +318,18 @@ def fill_sack(self, load_system_repo=True, load_available_repos=True):
                 errors = []
                 mts = 0
                 age = time.time()
+
+                repos_without_local_metadata = []
+                for r in self.repos.iter_enabled():
+                    try:
+                        r.load()
+                        if not r.metadata:
+                            repos_without_local_metadata.append(r)
+                    except dnf.exceptions.RepoError as e:
+                        self.handle_repo_error(r, errors, e)
+                self.download_metadata(repos_without_local_metadata)
+
+                dnf.util._terminal_messenger('print', _("Loading repositories."), sys.stdout)
```

Comment:
This line will be problematic with repoquery, repolist, ... where stdout is reserved for command output